hs_core/views/resource_rest_api.py

```diff
@@ -148,33 +162,36 @@ def get_queryset(self):
         for resources in resource_table.values():
             res = res.union(resources)
 
-        res_list = list(res)
-        for res in res_list:
-            resource_bag = hydroshare.get_resource(res.short_id)
-            if res.creator.first_name:
-                creator_name = res.creator.first_name
-                if res.creator.last_name:
-                    creator_name += " %s" % res.creator.last_name
-            else:
-                creator_name = res.creator.username
-
-            sharing_status = 'Public' if res.public else 'Private'
-
-            bag_url = hydroshare.utils.current_site_url() + resource_bag.bag.url
-            science_metadata_url = hydroshare.utils.current_site_url() + reverse('get_update_science_metadata', args=[res.short_id])
-            resource_list_item = serializers.ResourceListItem(resource_type=res.__class__.__name__,
-                                                              resource_id=res.short_id,
-                                                              resource_title=res.title,
-                                                              creator=creator_name,
-                                                              sharing_status=sharing_status,
-                                                              date_created=res.created,
-                                                              date_last_updated=res.updated,
-                                                              bag_url=bag_url,
-                                                              science_metadata_url=science_metadata_url)
+        for r in res:
+            resource_list_item = self._resourceToResourceListItem(r)
             filtered_res_list.append(resource_list_item)
 
         return filtered_res_list
 
+    def _resourceToResourceListItem(self, r):
+        resource_bag = hydroshare.get_resource(r.short_id)
+        if r.creator.first_name:
+            creator_name = r.creator.first_name
+            if r.creator.last_name:
+                creator_name += " %s" % r.creator.last_name
+        else:
+            creator_name = r.creator.username
+
+        sharing_status = 'Public' if r.public else 'Private'
```

Comment:
@ironfroggy  I'm happy to make this change.  @pkdash Do you have any objections?